### PR TITLE
Gamepad functionality

### PR DESCRIPTION
**Pull request by gitautas**
Added virtual gamepad, rudimentary unit tests, some light changes.

### CODE REVIEW

**Comment by gitautas** [comment]
This is linked to #27

**Comment by AndrusGerman** [comment]
The long-awaited moment, I'm trying this <3

**Comment by bendahl** [comment]
@gitautas Thanks for putting so much effort into this! So far, the changes are looking good. I first thought that putting the new implementation into the newly created "exp" package instead of using "uinput" would be preferable, as it would leave more freedom to evolve the API. However, looking at the current API, this is looking pretty good. Putting this in the uinput package is a good choice. 

Can you please clarify whether the axis events might contain a subtle bug (see my inline comment)?

**Comment by gitautas** [comment]
@bendahl can you confirm that you've left the comment, I can't see it.

**Comment by bendahl** [comment]
@gitautas Yes, just checked. The comment is located in gamepad.go. Please double check lines 211 ff. It looks to me like only one axis is altered (maybe a copy and paste error). 

Just as a heads-up: I don't have access to a computer at the moment, so I have to rely on the GitHub app to review this...

File: gamepad_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
+package uinput_test
+
+import (
+	"testing"
+	"time"
+	"uinput"
```
Comment (AndrusGerman) [inline gamepad_test.go:6]: My tests to this file don't pass, and the only way I can find to make this work is to change this import to
`"github.com/bendahl/uinput"`

**Review by AndrusGerman — CHANGES_REQUESTED** [review]
When running the unit tests, I don't compile the package. There seems to be a bug in the name of your import

File: gamepad_test.go (hunk 1 above)
Comment (gitautas) [reply gamepad_test.go:6]: Which version of golang are you using?

**Comment by gitautas** [comment]
> Just as a heads-up: I don't have access to a computer at the moment, so I have to rely on the GitHub app to review this...

I'm not crazy! You didn't leave a comment! I think I noticed the bug you mentioned so I'll fix it in a second.

File: gamepad_test.go (hunk 1 above)
Comment (AndrusGerman) [reply gamepad_test.go:6]: go version go1.18.3 linux/amd64, I also add that I am working with go modules

**Comment by bendahl** [comment]
@gitautas I finally got around to test this out locally and also run the CI/CD pipeline on it. Both failed due to the same errors that @AndrusGerman mentioned. I had a closer look at this and it seems that the issue is the usage of black box testing (uinput_test package ifor tests instead of uinput). I understand the intent, but would prefer it if we could stick to simple whitebox-tests. The reason for this is that this way we're easily able to test internal functions quickly without much fuss. 

In addition, for the CI/CD not to go nuts, there can be no infinite loops within the tests. For the sake of simply verifying that the events are actually fired as intended, it would even be sufficient to do away with the loops altogether and only have a single run (if it works once, it should work on subsequent invocations as well). This would also spare time spent on tests and not burn through the CI/CD-minutes. 

So, to sum things up, please change the following two things in your tests:

1. Use the uinput package instead of uinput_test.
2. Get rid of the loops (these are surely useful for manual testing, but a hindrance for CI/CD).

Other than that I don't see any further changes that would be necessary. 

Thanks again for all the effort and your patience! I really appreciate this!

Cheers
Ben

**Comment by gitautas** [comment]
@bendahl pinging you to remind that I added the requested changes.
